tests/test_scene_optimizer_no_gt.py

```diff
@@ -0,0 +1,97 @@
+"""Unit tests for the scene-optimizer class.
+
+Authors: Ayush Baid, John Lambert
+"""
+import unittest
+from pathlib import Path
+
+import dask
+import hydra
+import numpy as np
+from dask.distributed import Client, LocalCluster
+from gtsam import EssentialMatrix, Rot3, Unit3
+from hydra.utils import instantiate
+
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.loader.olsson_loader import OlssonLoader
+from gtsfm.retriever.exhaustive_retriever import ExhaustiveRetriever
+from gtsfm.retriever.retriever_base import ImageMatchingRegime
+from gtsfm.scene_optimizer import SceneOptimizer
+
+DATA_ROOT_PATH = Path(__file__).resolve().parent / "data"
+
+
+class TestSceneOptimizerNoGT(unittest.TestCase):
+    """Unit test for SceneOptimizer without any ground truth data available, which runs SfM for a scene."""
+
+    def setUp(self) -> None:
+        self.loader = OlssonLoader(
+            str(DATA_ROOT_PATH / "set3_lund_door_nointrinsics_noextrinsics"), image_extension="JPG"
+        )
+        assert len(self.loader)
+
+    def test_create_computation_graph(self):
```

Comment:
is the logic here identical to `test/test_scene_optimizer:test_create_computation_graph`?

Could we just parameterize the other function with
```
self.loader = OlssonLoader(
            str(DATA_ROOT_PATH / "set3_lund_door_nointrinsics_noextrinsics"), image_extension="JPG"
        )
```
to avoid code duplication?

Reply:
@ayushbaid marking this as resolved, thanks for consolidating the code here, much cleaner now!